src/jit/lsrabuild.cpp

```diff
@@ -1581,6 +1581,13 @@ void LinearScan::buildRefPositionsForNode(GenTree* tree, BasicBlock* block, Lsra
                 }
             }
         }
+
+        if (tree->OperIsPutArgSplit())
+        {
+            // If it is a putArgSplit then it can have previous PUTARG_REG nodes that can't be split.
```

Comment:
Did you mean "can't be spilled" rather than "can't be split"?
Also, this code generally assumes that any register not in use by the current node can be spilled. Do you know why it's the case that r0-r2 can't be? (I.e they would be spilled here, and then reloaded at the call - arguably ugly and undesirable, but that's sort of the point of these stress modes.)

Reply:
> Do you know why it's the case that r0-r2 can't be? 

Because they are special args that are marked as `isBusyUntilNextKill` and `isSpillCandidate` returns false for them:
https://github.com/dotnet/coreclr/blob/67ed270f13012e85b2925360d32673614f834449/src/jit/lsra.cpp#L3303-L3306.

> Did you mean "can't be spilled" rather than "can't be split"?

Fixed, thank you.

Reply:
> Because they are special args that are marked as isBusyUntilNextKill and isSpillCandidate returns false for them:

Ah, I see the flaw. The loop above increments minRegCount for the `isSpecialPutArg` nodes, and that was intended to account for this case (those are the ones that *might* be marked `isBusyUntilNextKill`). However, because it's only look at the `RefPosition`s that are built for this node, it won't actually see the outstanding `isSpecialPutArg` defs that aren't part of this node. So it would seem to me that a "correct" fix (though again one wonders about the benefit of finely tuning this stress-only code), would be to separately track the `isSpecialPutArg` intervals in flight, and always add that to the minRegCount.

At first I was concerned that this could impact `GT_PUTARG_REG` nodes in general - but of course they will have a specific register requirement, which can't conflict with any previous `isSpecialPutArg` nodes. And `GT_PUTARG_STK` nodes should never appear after any of the `GT_PUTARG_REG` nodes, so I think that only `GT_PUTARG_SPLIT` nodes would have this issue.